### PR TITLE
Implement AWS approved trust-policy hardening executor (#1539)

### LINKED ISSUE #1539: AWS Platform Wave 8.04: Approved trust policy hardening executor
## Parent epic
Part of #1472.

## Wave number and sequence number
Wave 8, issue 04.

## Problem statement
- Approved remediation needs trust policy hardening executor so Identrail can move from recommendations to controlled, auditable fixes.
- Without this, live changes either remain manual or become unsafe at scale.
- This issue adds the safety-controlled execution path required for verified AWS identity remediation.
- Operators need this issue to specify exact entities, API behavior, app behavior, evidence boundaries, and failure states so the PR can be implemented without guessing or silently widening scope.

## Why this matters for Identrail's AWS machine identity platform
- This issue is one PR-sized step in the AWS machine identity loop: connect, discover, normalize, graph, observe runtime, reason, recommend, approve, remediate, verify, and govern.
- It is required so Identrail can become an AWS machine-identity control plane, not another static scanner.
- The output must be explainable to operators with evidence, confidence, tradeoffs, and visible app state.

## Exact implementation scope
- Apply approved trust-policy hardening plans through controlled IAM update paths.
- Validate simulation/preconditions and record safe before/after policy docs.
- Run verification and provide rollback record.
- Wire the implementation into the relevant API and app path with operator-visible status, evidence links, confidence/failure handling, and filters or drill-downs required for this capability.

## Backend/API scope
- Add controlled execution services for trust policy hardening executor.
- Persist immutable audit entries, state transitions, idempotency keys, dry-run/apply/verify outputs, and rollback metadata.
- Enforce approval, RBAC, feature flags, and scope checks server-side.

## App/UI scope
- Add/update AWS app surface so operators can see trust policy hardening executor without reading logs or database rows.
- Include loading, empty, error, degraded, and permission-denied states.
- Show evidence links, confidence, timestamps, account/region context, and next actions.
- Show approval, dry-run, execution, verification, rollback, and audit states.

## Data model or graph contract impact
- Add plan/case/action records for trust policy hardening executor.
- Track lifecycle, actors, dry-run/apply/verify outputs, rollback metadata, and immutable audit.
- Redact sensitive policy/secret material as needed.

## Collector/runtime/remediation/governance impact where relevant
- This remediation capability makes trust policy hardening executor available to downstream AWS identity graph, runtime, reasoning, remediation, governance, and app flows as applicable.
- Preserve read-only behavior unless this issue explicitly implements approved remediation or governance execution.

## Tests required
- Unit tests for trust policy hardening executor mapping, validation, and failure behavior.
- Fixture/contract tests for normalized payload and graph/API output.
- Regression tests for empty, partial failure, permission denied, and degraded states.
- App/API integration tests for operator-visible workflows and filters.
- Security tests for RBAC, approvals, feature flags, idempotency, and kill-switch behavior.

## Documentation required
- Document what trust policy hardening executor implements, prerequisites, expected outputs, and limitations.
- Document AWS permissions and what data is intentionally not collected.
- Document live validation and troubleshooting.
- Document safety gates, tradeoffs, rollback/verification, and audit expectations.

## Safety/security constraints
- Default to read-only AWS access unless this issue explicitly implements approved remediation or controlled governance execution.
- Never read, expose, log, or persist secret values, prompt contents, completions, browser pages, code-interpreter output, database rows, object contents, or customer payloads by default.
- Preserve tenant, workspace, project, connector, account, and region boundaries in every API, worker, app, and graph path.
- Treat unknown, permission-denied, unsupported, and partially failed states as explicit states, not as successful findings.
- Do not create public security-vulnerability details; keep issue and PR language focused on platform capability and safe implementation.

## Live validation steps
- Start from the current `origin/dev` state and confirm blockers are closed before opening the PR.
- Run fixture validation for this issue and confirm normalized records/API output match the contract.
- Validate the app surface for success, empty, degraded, permission-denied, and partial-failure states where applicable.
- For any live AWS validation, use an authorized test account only and record account/region/service coverage without exposing sensitive payloads.
- Include evidence screenshots, API examples, or command output summaries in the PR notes.

## Definition of done
- The exact scope above is implemented as one focused PR with no unrelated AWS expansion.
- Backend/API behavior, graph/data contracts, app surfaces, tests, docs, and safety constraints are all complete for this issue.
- Operators can tell what was collected, inferred, planned, executed, or governed; why it matters; what evidence supports it; and what is intentionally out of scope.
- Failure modes are explicit and do not create false positives.
- The PR follows `CONTRIBUTING.md`, references this issue, and closes only this focused capability.

## Explicit out of scope
- Do not implement later-wave capabilities listed as blockers or downstream consumers.
- Do not add live AWS mutation unless this issue is specifically an approved remediation/governance executor.
- Do not collect secret values, customer payloads, prompts, completions, browser output, database contents, or object contents.
- Do not broaden this issue into a general AWS scanner or unrelated cloud/provider work.

## Dependencies/blockers
- Blocked by: #1537. Do not open a PR for this issue until #1537 is closed.
- Blocked by: #1531. Do not open a PR for this issue until #1531 is closed.

### PR DESCRIPTION
**Pull request by Oluwatobi-Mustapha**
## Summary

Adds a read-only projection that joins the dry-run executor (#1537) with the
trust-policy hardening planner (#1531) for cases whose `source_type` is
`trust_policy_hardening` and whose dry-run diff kind targets an IAM
trust-policy mutation (`iam_trust_diff` or `iac_trust_policy_pr`). Each entry
copies the deterministic idempotency key, the intended
`iam:UpdateAssumeRolePolicy` call, the planner's structured trust-policy
fields, policy-simulator metadata, seven preconditions, verification records,
rollback plan, and an audit trail with a
`trust_policy_hardening_execution_projected` row.

State semantics: `projected` (every precondition passes + dry-run ready),
`precondition_failed` (planner not yet ready / breakage above low / dry-run
not yet `would_succeed`), or `blocked` (kill switch engaged, public principal
would remain, idempotency key missing, or upstream is itself blocked).
`ready_for_live_apply` is true only when state is `projected`, the upstream
dry-run is itself `ready_for_apply`, and no kill switch is engaged.

Closes #1539. Part of #1472.

## Why

Wave 8 needs a focused executor for the trust-policy hardening pathway,
joining dry-run readiness with the planner's structured trust-policy fields
so the wave-8 apply runtime can replay the change idempotently when its
feature flag opens.

## Scope

- [x] Focused change (no unrelated modifications)
- [x] Backward compatibility considered (additive endpoint, types, and UI)
- [x] Risk level assessed (read-only projection; never calls AWS write APIs; never inlines policy bodies or secrets)

## Validation

\`\`\`bash
make fmt-check
go vet ./...
go test ./internal/api/...
make api-example-contract-check
make web-route-integrity-check
npm run test:ci --prefix web
npm run build --prefix web
\`\`\`

All passed locally on this branch (fresh off origin/dev).

## Checklist

- [x] Tests added/updated for behavior changes
- [x] Docs updated for user/operator-facing changes
- [x] \`CHANGELOG.md\` updated when relevant
- [x] No secrets, credentials, or sensitive data included

## AI Assistance Disclosure

- AI-assisted: no
- Tools used:
- Areas where used:
- Human verification performed:

## Related Issues

Closes #1539
Refs #1472

---
## Summary by cubic
Adds an AWS approved trust-policy hardening executor: a read-only projection that joins the remediation dry-run executor with the trust-policy hardening planner so operators can review idempotent, ready-for-apply trust-policy changes without calling AWS write APIs. Fulfills Wave 8 executor requirements for #1539 and supports #1472.

- **New Features**
  - API: `GET /v1/workspaces/:workspace_id/projects/:project_id/aws/trust-policy-hardening-executor` with filters for connector, fixture_state, account, region, dry-run ID, case ID, plan ID, hardening direction, state, severity, and search.
  - Projection: idempotency key, intended `iam:UpdateAssumeRolePolicy` call, planner fields (direction, principal change, condition recs/signals/snippets, affected callers, breakage), policy-simulator metadata, preconditions, verifications, rollback plan, and an audit trail.
  - State and safety: `projected`, `precondition_failed`, `blocked`. `ready_for_live_apply` only when projected, upstream dry-run is ready_for_apply, and no kill switch is engaged. Metadata-only; never calls AWS write APIs or exposes rendered policy bodies or secrets.
  - Wiring: added route to the built-in policy bundle and OpenAPI; server with filtering/relationships/diagnostics; docs; web API client + tests and a new UI panel.

- **Bug Fixes**
  - Fixed dry-run source joins for trust-policy hardening and restricted cases to IAM roles.
  - Prioritized kill-switch and other safety blocks; entries marked `blocked` when any guard trips; fixed approval readiness.
  - Policy-simulator now uses planner condition signals; preserved upstream dry-run diagnostics.
  - Stabilized trust evidence: deterministic corroboration base, recomputed status after dedupe, preserved unconditioned evidence; retained IAM last-used partial fixtures.
  - Updated executor validation to respect the planner’s hardening direction when checking readiness and outcomes.

<sup>Written for commit 1b2d9f1244f95a0b963cd486803f688c0cfcaf7e. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/identrail/identrail/pull/1686?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

File: internal/api/aws_trust_policy_hardening_executor.go [hunk 1]
```diff
@@ -0,0 +1,773 @@
...
+		entries = append(entries, awsTrustPolicyHardeningExecutorEntryFromDryRun(entry, plan, now))
+	}
+	return entries
+}
+
+// awsTrustPolicyHardeningExecutorAdmits guards the executor to the
+// `trust_policy_hardening` source type AND an IAM trust-policy diff kind so
+// non-trust dry-runs never bleed into this projection.
+func awsTrustPolicyHardeningExecutorAdmits(entry AWSRemediationDryRunEntry) bool {
+	if !strings.EqualFold(entry.SourceType, "trust_policy_hardening") {
```
[the remark this answers is not in the document]
Comment (Oluwatobi-Mustapha) [reply internal/api/aws_trust_policy_hardening_executor.go:329]: Fixed in cac1cec8. The remediation-case producer now emits trust-policy hardening planner records as `trust_policy_hardening` cases, so the approval and dry-run pipeline produces the source type the executor accepts. I also tightened the remediation-case and executor contract tests so the success fixture must include trust-policy cases and non-empty executor entries.

File: internal/api/aws_trust_policy_hardening_executor.go [hunk 2]
```diff
@@ -0,0 +1,773 @@
...
+		return awsTrustPolicyHardeningExecutorStateBlocked
+	}
+	for _, precondition := range preconditions {
+		if precondition.Status != "blocked" {
+			continue
+		}
+		if awsTrustPolicyHardeningExecutorPreconditionIsSafety(precondition.Name) {
+			return awsTrustPolicyHardeningExecutorStateBlocked
+		}
+		return awsTrustPolicyHardeningExecutorStatePreconditionFailed
```
[the remark this answers is not in the document]
Comment (Oluwatobi-Mustapha) [reply internal/api/aws_trust_policy_hardening_executor.go:502]: Fixed in 4612e656. The executor state reducer now scans all blocked preconditions before returning, so any safety gate such as `no_public_principal_after_change` takes priority over retryable readiness failures. I also added a regression assertion for a plan that is both not ready and still has a public principal.

File: internal/api/aws_remediation_cases.go [hunk 3]
```diff
@@ -580,6 +590,62 @@ func awsRemediationCaseFromBlastRadius(finding AWSBlastRadiusFinding, now time.T
...
+		Region:             plan.Region,
+		IdentityNodeID:     plan.ResourceNodeID,
+		IdentityARN:        plan.ResourceARN,
+		IdentityName:       firstNonEmptyAWSValue(plan.ResourceLabel, shortAWSARN(plan.ResourceARN), plan.ResourceNodeID),
+		IdentityType:       "iam_role",
+		ResourceNodeIDs:    awsRemediationResourceNodes(plan.ResourceNodeID, plan.ImpactedNodes),
+		Owner:              owner,
+		OwnerAssigned:      ownerAssigned,
+		ApprovalRequired:   approvalRequired,
+		ApprovalState:      approvalState,
```
[the remark this answers is not in the document]
Comment (Oluwatobi-Mustapha) [reply internal/api/aws_remediation_cases.go:640]: Fixed in 3c57fc6d. Ready trust-policy hardening plans now preserve `approval_state=approved` when the case is created, so the approval queue no longer demotes them to `under_review` before the dry-run/executor path can project readiness. I added a regression test for a ready trust-policy plan keeping the approved lifecycle/state.

File: internal/api/aws_trust_policy_hardening_executor.go [hunk 4]
```diff
@@ -0,0 +1,777 @@
...
+	return awsTrustPolicyHardeningExecutorStateProjected
+}
+
+// awsTrustPolicyHardeningExecutorPreconditionIsSafety names the precondition
+// checks that put the entry into `blocked`. Other failures (planner not yet
+// ready, breakage level not yet low) are surfaced as `precondition_failed`
+// so operators can see the difference between "unsafe" and "not yet ready".
+func awsTrustPolicyHardeningExecutorPreconditionIsSafety(name string) bool {
+	switch name {
+	case "kill_switch_off", "idempotency_key_present", "upstream_prerequisites", "no_public_principal_after_change":
```
[the remark this answers is not in the document]
Comment (Oluwatobi-Mustapha) [reply internal/api/aws_trust_policy_hardening_executor.go]: Fixed in 3c57fc6d. `upstream_prerequisites` is no longer treated as a safety precondition by itself; actual upstream blocked/kill-switched outcomes still return `blocked`, while pending approval prerequisites now remain retryable `precondition_failed`. I added a regression assertion for the pending approval case.

File: internal/api/aws_trust_policy_hardening_executor.go [hunk 5]
```diff
@@ -0,0 +1,777 @@
...
+		Signals:       signals,
+	}
+}
+
+func awsTrustPolicyHardeningExecutorVerifications(entry AWSRemediationDryRunEntry, plan AWSTrustPolicyHardeningPlan) []AWSTrustPolicyHardeningExecutorVerification {
+	out := []AWSTrustPolicyHardeningExecutorVerification{
+		{Source: "cloudtrail", Signal: "expected_api_call_observed", Status: "pending", Description: "After live execution, confirm iam:UpdateAssumeRolePolicy appears in CloudTrail for the target account and region."},
+		{Source: "access_analyzer", Signal: "no_new_external_findings", Status: "pending", Description: "Re-run Access Analyzer after live execution to confirm no new external-trust findings."},
+	}
+	if strings.EqualFold(plan.HardeningDirection, "add_condition") {
```
[the remark this answers is not in the document]
Comment (Oluwatobi-Mustapha) [reply internal/api/aws_trust_policy_hardening_executor.go]: Fixed in 73c911ae. The executor now keys the `conditions_enforced` IAM simulator verification off the planner’s actual `ConditionRecommendations` instead of the non-emitted `add_condition` direction token. I added a regression test using the real `add_org_or_source_condition` planner direction.

File: internal/api/aws_remediation_cases.go [hunk 6]
```diff
@@ -580,6 +590,65 @@ func awsRemediationCaseFromBlastRadius(finding AWSBlastRadiusFinding, now time.T
...
 }
 
+func awsRemediationCaseFromTrustPolicyHardening(plan AWSTrustPolicyHardeningPlan, now time.Time) (AWSRemediationCase, bool) {
+	if plan.PlanID == "" {
+		return AWSRemediationCase{}, false
+	}
+	caseID := "aws-remediation-case:" + stableAWSBlastRadiusToken("trust-policy-hardening", plan.PlanID)
+	evidenceRef := firstString(awsRemediationEvidenceRefs(plan.Evidence))
+	diff := AWSRemediationDiffIntent{
+		Kind:               "iam_trust_diff",
```
[the remark this answers is not in the document]
Comment (Oluwatobi-Mustapha) [reply internal/api/aws_remediation_cases.go:605]: Fixed in 26ad23ac. The trust-policy hardening case producer now emits IAM trust diffs only for IAM role plans, so S3/KMS/Secrets/SQS/SNS/Dynamo/RDS resource-policy planner records no longer become UpdateAssumeRolePolicy dry-runs admitted by the executor. I added a regression test for non-IAM resource-policy plans being skipped and updated the executor contract test to assert the default resource-policy fixture stays out of the IAM executor.

File: internal/api/aws_remediation_cases.go [hunk 7]
```diff
@@ -321,7 +322,11 @@ func (s *Service) awsRemediationCaseSourceSignals(ctx context.Context, workspace
 	if err != nil {
 		return awsRemediationCaseSources{}, fmt.Errorf("remediation case blast radius: %w", err)
 	}
-	return awsRemediationCaseSources{risk: risk, least: least, equivalence: equivalence, blast: blast}, nil
+	trust, err := s.GetAWSTrustPolicyHardeningPlans(ctx, workspaceID, projectID, AWSTrustPolicyHardeningRequest{ConnectorID: connectorID, FixtureState: fixtureState})
```
[the remark this answers is not in the document]
Comment (Oluwatobi-Mustapha) [reply internal/api/aws_remediation_cases.go:327]: Fixed in 5cd71fa7. Remediation cases now include the trust-policy hardening source in status, failure reason, remediation hint, diagnostic, and coverage-gap aggregation, and the endpoint evidence links include the trust-policy hardening issue/docs. I added a regression test that a degraded trust source is surfaced by the remediation-case aggregators.

File: internal/api/aws_trust_policy_hardening_executor_test.go [hunk 8]
```diff
@@ -0,0 +1,340 @@
...
+	if err != nil {
+		t.Fatalf("get trust-policy hardening executor: %v", err)
+	}
+	if result.CurrentIssueRef != "#1539" || result.Version != awsTrustPolicyHardeningExecutorVersion {
+		t.Fatalf("unexpected contract metadata: %+v", result)
+	}
+	if result.Summary.RelationshipCount != len(result.Relationships) {
+		t.Fatalf("expected relationship count to match: summary=%+v relationships=%+v", result.Summary, result.Relationships)
+	}
+	if len(result.Entries) != 0 {
```
[the remark this answers is not in the document]
Comment (Oluwatobi-Mustapha) [reply internal/api/aws_trust_policy_hardening_executor_test.go]: Fixed in 5cd71fa7. The success fixture now includes a real cross-account STS AssumeRole event into an IAM role, so the trust-policy planner emits an IAM-role hardening plan and the executor success contract again exercises projected iam:UpdateAssumeRolePolicy entries. The non-IAM resource-policy guard remains in place.

File: internal/api/aws_runtime_events.go [hunk 9]
```diff
@@ -868,8 +870,12 @@ func awsRuntimeEventFixtureRecords(accountID string, region string, fixtureState
 			ExpiresAt:           started.Add(time.Hour),
 		}
 	}
+	crossAccountSession := session("sess-prod-to-runtime-reader", role, base.Add(4*time.Minute))
+	crossAccountSession.OriginalActorARN = productionRunnerRole
+	crossAccountSession.OriginalActorNodeID = awsIdentityNodeIDForAPI(productionRunnerRole)
 	records := []AWSRuntimeEventRecord{
 		awsRuntimeEventFixtureRecord(accountID, region, "evt-sts-runtime-reader", "sts-session", "sts.amazonaws.com", "AssumeRole", "sts:AssumeRole", role, "", "", "security", "cloudtrail", base, session("sess-runtime-reader", role, base)),
+		awsRuntimeEventFixtureRecord(accountID, region, "evt-sts-prod-runtime-reader", "sts-session", "sts.amazonaws.com", "AssumeRole", "sts:AssumeRole", productionRunnerRole, role, "iam_role", "security", "cloudtrail", base.Add(4*time.Minute), crossAccountSession),
```
[the remark this answers is not in the document]
Comment (Oluwatobi-Mustapha) [reply internal/api/aws_runtime_events.go:878]: Fixed in 49be2901. The partial_failure runtime fixture now filters out only Access Analyzer evidence instead of slicing by record position, so CloudTrail and IAM last-used signals remain visible after the added STS fixture. I added a regression test for fixture_state=partial_failure&event_type=iam-last-used returning the retained IAM last-used records.

File: internal/api/aws_trust_policy_hardening_executor.go [hunk 10]
```diff
@@ -0,0 +1,777 @@
...
+	sort.SliceStable(entries, func(i, j int) bool {
+		if entries[i].Score == entries[j].Score {
+			return entries[i].ExecutionID < entries[j].ExecutionID
+		}
+		return entries[i].Score > entries[j].Score
+	})
+	filtered, applied := filterAWSTrustPolicyHardeningExecutorEntries(entries, request)
+	relationships := awsTrustPolicyHardeningExecutorRelationships(filtered)
+	diagnostics := awsTrustPolicyHardeningExecutorDiagnostics(plans.Diagnostics)
+	coverageGaps := awsTrustPolicyHardeningExecutorCoverageGaps(plans.CoverageGaps)
```
[the remark this answers is not in the document]
Comment (Oluwatobi-Mustapha) [reply internal/api/aws_trust_policy_hardening_executor.go]: Fixed in 8470d470. The trust-policy hardening executor now merges dry-run diagnostics and coverage gaps with the planner diagnostics/gaps, so degraded or partial-failure dry-run upstreams surface their source details in the executor response. I added a regression test that preserves both dry-run and planner diagnostics/gaps.

File: internal/api/aws_runtime_events.go (hunk 9 above)
[the remark this answers is not in the document]
Comment (Oluwatobi-Mustapha) [reply internal/api/aws_runtime_events.go:878]: Fixed in f6c46237. The success fixture now includes Access Analyzer evidence for the same external principal/IAM role assumption, and cross-account trust merges runtime/analyzer corroboration for the same principal/resource pair while keeping the runtime IAM-role finding as the executor base. That gives the trust-policy hardening plan low breakage/ready_for_apply and restores projected ready executor rows for fixture_state=success&state=projected. I added regressions for ready IAM-role plans and projected ready executor entries.

File: internal/api/aws_cross_account_trust.go [hunk 11]
```diff
@@ -806,13 +806,104 @@ func awsCrossAccountTrustDedupeFindings(findings []AWSCrossAccountTrustFinding)
 		}
 		seen[finding.FindingID] = finding
 	}
-	out := make([]AWSCrossAccountTrustFinding, 0, len(seen))
+	corroborated := map[string]AWSCrossAccountTrustFinding{}
 	for _, finding := range seen {
```
[the remark this answers is not in the document]
Comment (Oluwatobi-Mustapha) [reply internal/api/aws_cross_account_trust.go]: Fixed in 2e6bc472. The corroboration merge no longer iterates maps to choose the surviving base/output order: it preserves explicit first-seen order, emits results through that ordered key list, and uses a deterministic FindingID tie-break for equal-score non-IAM-role matches. Runtime-observed IAM role findings still win over analyzer-only matches so the trust-policy executor keeps stable IAM-role plan/case/dry-run IDs. Added a regression that exercises equal-score corroboration in both input orders.

File: internal/api/aws_cross_account_trust.go [hunk 12]
```diff
@@ -797,22 +797,132 @@ func awsCrossAccountTrustFindingFromBlastRadius(finding AWSBlastRadiusFinding, a
...
+		base.ResourceARN = other.ResourceARN
+	}
+	if base.ResourceNodeID == "" {
+		base.ResourceNodeID = other.ResourceNodeID
+	}
+	if base.ResourceLabel == "" {
+		base.ResourceLabel = other.ResourceLabel
+	}
+	if base.Confidence < other.Confidence {
+		base.Confidence = other.Confidence
```
[the remark this answers is not in the document]
Comment (Oluwatobi-Mustapha) [reply internal/api/aws_cross_account_trust.go:890]: Fixed in ee7832a8. The corroboration merge now recomputes Status from the merged score/confidence after carrying analyzer confidence onto the runtime IAM-role base, so a low-confidence runtime role corroborated by high-confidence Access Analyzer evidence becomes action_required instead of staying review. I added a regression that walks the merged finding through trust-policy planning and remediation-case creation to confirm ReadyForApply keeps approval_state/lifecycle approved for the executor path.

File: internal/api/aws_cross_account_trust.go [hunk 13]
```diff
@@ -797,22 +797,133 @@ func awsCrossAccountTrustFindingFromBlastRadius(finding AWSBlastRadiusFinding, a
...
+func awsCrossAccountTrustMergeCorroboratingFindings(existing, incoming AWSCrossAccountTrustFinding) AWSCrossAccountTrustFinding {
+	base, other := existing, incoming
+	if awsCrossAccountTrustPrefersCorroborationBase(incoming, existing) {
+		base, other = incoming, existing
+	}
+	base.RuntimeObserved = base.RuntimeObserved || other.RuntimeObserved
+	base.AnalyzerBacked = base.AnalyzerBacked || other.AnalyzerBacked
+	base.PublicPrincipal = base.PublicPrincipal || other.PublicPrincipal
+	base.TrustedWithinOrganization = base.TrustedWithinOrganization || other.TrustedWithinOrganization
+	base.HasCondition = base.HasCondition || other.HasCondition
```
[the remark this answers is not in the document]
Comment (Oluwatobi-Mustapha) [reply internal/api/aws_cross_account_trust.go]: Fixed in ddf9496f. Corroboration now preserves the unsafe unconditioned signal: merged findings only report HasCondition=true when all merged evidence is conditioned, while still retaining condition keys from conditioned statements as evidence. I added a regression for same-principal/resource conditioned plus unconditioned grants that confirms the cross-account summary still counts the unconditioned grant and the trust-policy planner still chooses add_org_or_source_condition.

**Review by unicorn-iam — DISMISSED** [review]

File: docs/aws-trust-policy-hardening-executor.md [hunk 14]
```diff
@@ -0,0 +1,119 @@
...
+
+## Validation
+
+1. Confirm blockers #1531 and #1537 are closed.
+2. Run upstream dry-run + trust-hardening planner with `fixture_state=success`.
+3. Call the executor endpoint with `fixture_state=success` and verify at
+   least one entry includes intended API call, preconditions, policy
+   simulation, verifications, structured planner data, idempotency key,
+   and audit trail.
+4. Filter by `hardening_direction=add_condition`, `state=projected`, and
```
[the remark this answers is not in the document]
Comment (Oluwatobi-Mustapha) [reply docs/aws-trust-policy-hardening-executor.md]: Fixed in 1b2d9f12. The executor validation docs now use an emitted planner direction (`hardening_direction=add_org_or_source_condition`) instead of the stale `add_condition` alias, and the adjacent verification prose now refers generically to condition-hardening directions. I also rechecked the doc for remaining `add_condition` references.

**Review by unicorn-iam — APPROVED** [review]
